Sources/Core/AWSSDKHTTPAuth/Sources/AWSSDKHTTPAuth/AWSSigV4Signer.swift

```diff
@@ -256,12 +256,14 @@ public class AWSSigV4Signer: SmithyHTTPAuthAPI.Signer {
             }
         }
 
+        // STREAMING-UNSIGNED-PAYLOAD-TRAILER
+        if isUnsignedBody { return .streamingUnsignedPayloadTrailer }
+
         // streaming + eligible for chunked transfer
         if !checksumIsPresent {
```

Comment:
should we reverse this `if-else`?  Weird to have a negated condition for an `if-else`

Reply:
this whole if-else can now break down to one statement BTW

Reply:
Changed as suggested.